### PR TITLE
Changes faraday_middleware dependency to < 0.9

### PR DESCRIPTION
**Pull request by hankwallace**
For this gem to be used in Backend, the `faraday_middleware` dependency has to be `< 0.9`.

There are older versions of this gem that have a compatible `faraday_middleware` dependency, but they are not JSON API v1 compatible.

I posted a question in the Dev slack room regarding versioning. The next patch of this would be `1.0.2`, so that's what I made it here. Does it matter that the upstream/source gem will probably have the same version in the future?

### CODE REVIEW

File: test/unit/destroying_test.rb [hunk 1]
```diff
@@ -21,14 +21,16 @@ def test_destroy
...
-  def test_destroy_no_content
-    stub_request(:delete, "http://example.com/users/6")
-      .to_return(headers: {content_type: "application/vnd.api+json"}, body: nil)
-
-    user = User.new(id: 6)
-    assert(user.destroy, "successful deletion should return truish value")
-    assert_equal(false, user.persisted?)
-  end
+  # Commenting this out for now. The faraday_middleware < 0.9 gem doesn't properly
+  # handle nil responses.
```
Comment (justin-yesware) [inline test/unit/destroying_test.rb]: What does it do with `nil` responses? I think using this test to document the expected behavior would be a great idea.

Comment (justin-yesware) [inline test/unit/destroying_test.rb]: Likewise with the commented out test below.

Comment (hankwallace) [inline test/unit/destroying_test.rb]: It will generate an exception. These tests were added at the same time as the `faraday_middleware` upgrade.

**Comment by hankwallace** [comment]
Closing this for now. I think I can use the upstream/source instead.

**Pull request closed without merging**